src/ui/components/EamlightToolbar.js

```diff
@@ -138,41 +142,80 @@ class EamlightToolbar extends Component {
     //
     getRegions = () => {
         const { regions, toggleHiddenRegion, getUniqueRegionID, isHiddenRegion } = this.props;
-        return regions
-            .filter(region => !region.ignore)
-            .map(region => (
-                <MenuItem key={region.id} onClick={() => toggleHiddenRegion(getUniqueRegionID(region.id))}>
-                    <Checkbox disabled checked={!isHiddenRegion(region.id)}/>
-                    {region.label}
-                </MenuItem>
-        ))
+    
+        let tabs = {};
+    
+        for (const region of regions) {
+            const category = region.menuCategory || DEFAULT_MENU_CATEGORY;
+    
+            if (!tabs[category]) {
+                tabs[category] = [];
+            }
+    
+            if (!region.ignore) {
+                let item = (
+                    <MenuItem key={region.id} onClick={() => toggleHiddenRegion(getUniqueRegionID(region.id))}>
+                        <Checkbox disabled checked={!isHiddenRegion(region.id)} />
+                        {region.label}
+                    </MenuItem>
+                );
+                tabs[category].push(item);
+            }
+        }
+    
+        return (
+            <>
+                {Object.keys(tabs).map((category, index) => (
```

Comment:
Using the indexof an array as a key is generally not recommended. Instead, you can use a single key for each element (category in this case)